addons/mail/static/tests/discuss_app/discuss.test.js

```diff
@@ -772,6 +780,76 @@ test("rendering of inbox message", async () => {
     await contains(".o-dropdown-item:contains('Translate')");
 });
 
+test("Can right-click on message to opens message actions dropdown", async () => {
+    const pyEnv = await startServer();
+    const partnerId = pyEnv["res.partner"].create({ name: "Refactoring" });
+    const [messageId_1, messageId_2] = pyEnv["mail.message"].create([
+        {
+            body: "message-body-1",
+            model: "res.partner",
+            needaction: true,
+            res_id: partnerId,
+        },
+        {
+            body: "msg-body-2",
+            model: "res.partner",
+            needaction: true,
+            res_id: partnerId,
+        },
+    ]);
+    pyEnv["mail.notification"].create([
+        {
+            mail_message_id: messageId_1,
+            notification_status: "sent",
+            notification_type: "inbox",
+            res_partner_id: serverState.partnerId,
+        },
+        {
+            mail_message_id: messageId_2,
+            notification_status: "sent",
+            notification_type: "inbox",
+            res_partner_id: serverState.partnerId,
+        },
+    ]);
+    await start();
+    await openDiscuss("mail.box_inbox");
+    await contains(".o-mail-Message", { count: 2 });
+    await rightClick(".o-mail-Message:eq(0)");
+    await contains(".o-dropdown-item", { count: 6 });
+    await contains(".o-dropdown-item:contains('Add a Reaction')");
+    await contains(".o-dropdown-item:contains('Add Star')");
+    await contains(".o-dropdown-item:contains('Mark as Read')");
+    await contains(".o-dropdown-item:contains('Reply')");
+    await contains(".o-dropdown-item:contains('Translate')");
+    await contains(".o-dropdown-item:contains('Copy Text')");
+    await contains(".o-mail-Message:eq(0).o-selected");
+    await contains(".o-mail-Message:eq(1):not(.o-selected)");
```

Comment:
Isn't contains deprecated? We might want to use the new thing when introducing new tests

Reply:
We would want to use web helpers, but they are still not ready, e.g. lack match of `text` (`/^$/` regex sucks).
Also I made some quick experiment to convert few `contains()` to web equivalent and saw the web helpers were about 25% slower. We have internal complains the HOOT test suite on runbot is reaching the time limit in `master`, so 25% is a big deal.